stratosphere/pm/source/boot2/boot2_api.cpp

```diff
@@ -0,0 +1,294 @@
+            return R_SUCCEEDED(gpioPadGetValue(&button, &val)) && val == GpioValue_Low;
+        }
+
+        bool IsMaintenanceMode() {
+            /* Contact set:sys, retrieve boot!force_maintenance. */
+            DoWithSmSession([&]() {
+                R_ASSERT(setsysInitialize());
+            });
+            {
+                ON_SCOPE_EXIT { setsysExit(); };
+
+                u8 force_maintenance = 1;
+                setsysGetSettingsItemValue("boot", "force_maintenance", &force_maintenance, sizeof(force_maintenance));
+                if (force_maintenance != 0) {
+                    return true;
+                }
+            }
+
+            /* Contact GPIO, read plus/minus buttons. */
+            DoWithSmSession([&]() {
+                R_ASSERT(gpioInitialize());
+            });
+            {
+                ON_SCOPE_EXIT { gpioExit(); };
+
+                return GetGpioPadLow(GpioPadName_ButtonVolUp) && GetGpioPadLow(GpioPadName_ButtonVolDown);
+            }
+        }
+
+        void WaitForMitm(const char *service) {
+            const auto name = sts::sm::ServiceName::Encode(service);
+
+            while (true) {
+                bool mitm_installed = false;
+                R_ASSERT(sts::sm::mitm::HasMitm(&mitm_installed, name));
+                if (mitm_installed) {
+                    break;
+                }
+                svcSleepThread(1'000'000ull);
+            }
+        }
+
+        void LaunchFlaggedProgramsFromSdCard() {
+            /* Allow for user-customizable programs. */
+            DIR *titles_dir = opendir("sdmc:/atmosphere/titles");
+            struct dirent *ent;
+            if (titles_dir != NULL) {
+                while ((ent = readdir(titles_dir)) != NULL) {
+                    if (strlen(ent->d_name) == 2 * sizeof(u64) && IsHexadecimal(ent->d_name)) {
+                        ncm::TitleId title_id{strtoul(ent->d_name, NULL, 16)};
+                        if (pm::info::HasLaunchedTitle(title_id)) {
+                            return;
+                        }
+                        char title_path[FS_MAX_PATH];
+                        std::snprintf(title_path, sizeof(title_path), "sdmc:/atmosphere/titles/%s/flags/boot2.flag", ent->d_name);
+                        FILE *f_flag = fopen(title_path, "rb");
+                        if (f_flag != NULL) {
+                            fclose(f_flag);
+                            LaunchTitle(nullptr, ncm::MakeTitleLocation(title_id, ncm::StorageId::None), 0);
+                        }
+                    }
+                }
+                closedir(titles_dir);
+            }
+        }
+
+    }
+
+    /* Boot2 API. */
+    void LaunchBootPrograms() {
+        /* Wait until fs.mitm has installed itself. We want this to happen as early as possible. */
+        WaitForMitm("fsp-srv");
+
+        /* Launch programs required to mount the SD card. */
+        LaunchList(PreSdCardLaunchPrograms, NumPreSdCardLaunchPrograms);
```

Comment:
Agree with `std::size` sentiment above.
```suggestion
        LaunchList(PreSdCardLaunchPrograms, std::size(PreSdCardLaunchPrograms));
```

Reply:
Yep.